### PR TITLE
JBIDE-27611: Provides devfile code assist and validation

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by jeffmaury** [comment]
testPR

**Comment by jeffmaury** [comment]
testPR

File: plugins/org.jboss.tools.openshift.core/plugin.xml [hunk 1]
```diff
@@ -365,5 +365,22 @@
...
+          id="org.jboss.tools.openshift.core.devfile"
+          name="Devfile"
+          priority="normal">
+    </content-type>
+ </extension>
+ <extension
+       point="org.eclipse.wildwebdeveloper.json.schema">
+    <schema
+          pattern="devfile.y*ml"
+          url="platform:/plugin/org.jboss.tools.openshift.core/devfile.json">
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.core/plugin.xml:382]: shouldn't be 
```suggestion
          url="platform:/plugin/org.jboss.tools.openshift.core/resources/devfile.json">
```
i get error message on devfile.yaml saying that json is not reachable

**Review by sbouchet — CHANGES_REQUESTED** [review]

File: plugins/org.jboss.tools.openshift.core/plugin.xml (hunk 1 above)
Comment (jeffmaury) [reply plugins/org.jboss.tools.openshift.core/plugin.xml:382]: No location is correct as the file is at the root of the generated JAR. There may be something missing for Eclipse to find the file in dev mode

**Review by sbouchet — APPROVED** [review]
LGTM after review discussion

**Pull request merged**